android/engine/src/main/java/org/smartregister/fhircore/engine/data/local/DefaultRepository.kt

```diff
@@ -681,18 +682,38 @@ constructor(
       }
       is CarePlan -> resource.status = CarePlan.CarePlanStatus.COMPLETED
       is Procedure -> resource.status = Procedure.ProcedureStatus.STOPPED
-      is Condition ->
-        resource.clinicalStatus =
-          CodeableConcept().apply {
-            coding =
-              listOf(
-                Coding().apply {
-                  system = SNOMED_SYSTEM
-                  display = PATIENT_CONDITION_RESOLVED_DISPLAY
-                  code = PATIENT_CONDITION_RESOLVED_CODE
-                }
-              )
+      is Condition -> {
+        if (resourceConfig.id == PNC_CONDITION_TO_CLOSE_RESOURCE_ID) {
+          val closePncCondition = resource.onset.dateTimeValue().value.daysPassed() > 28
+          if (closePncCondition) {
+            resource.clinicalStatus =
+              CodeableConcept().apply {
+                coding =
+                  listOf(
+                    Coding().apply {
+                      system = SNOMED_SYSTEM
+                      display = PATIENT_CONDITION_RESOLVED_DISPLAY
+                      code = PATIENT_CONDITION_RESOLVED_CODE
+                    }
+                  )
+              }
           }
+        } else {
+          resource.clinicalStatus =
+            CodeableConcept().apply {
+              coding =
+                listOf(
+                  Coding().apply {
+                    system = SNOMED_SYSTEM
+                    display = PATIENT_CONDITION_RESOLVED_DISPLAY
+                    code = PATIENT_CONDITION_RESOLVED_CODE
+                  }
+                )
+            }
+        }
+
```

Comment:
is the hard coding we'll remove somewhere here? can we add a todo whereever that is

Reply:
cool thx

Reply:
cool thx